### PR TITLE
introduce defaults

### PR DESCRIPTION
**Pull request by Karsten1987**
This PR sits on top of #451 to show case the difference when decent defaults are chosen.
I'd like to hear from others whether these defaults are sensible or not.

### CODE REVIEW

**Comment by Karsten1987** [comment]
@emersonknapp sorry, I've added one other thing to this PR. I introduce an overload for `open` for reader and writer which take a single `uri` string to a rosbag and internally fills the `storage_options`. Given that we internally set the storage options to the default, which is `sqlite3`, I believe this introduces a `exec_depend` on `rosbag2_storage_default_plugins`. I do think that's a reasonable thing to do though. But I'd like to hear your opinion about it, whether that's sensible to introduce as a dependency.

**Comment by ros-discourse** [comment]
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/ros-2-tsc-meeting-minutes-2020-07-16/15468/1

File: rosbag2_cpp/src/rosbag2_cpp/reader.cpp [hunk 1]
```diff
@@ -34,6 +35,18 @@ Reader::~Reader()
   reader_impl_->reset();
 }
 
+void Reader::open(const std::string & uri)
+{
+  rosbag2_cpp::StorageOptions storage_options;
+  storage_options.uri = uri;
+  storage_options.storage_id = "sqlite3";
```
Comment (emersonknapp) [inline rosbag2_cpp/src/rosbag2_cpp/reader.cpp:42]: minor nit - should these 3 default values ("sqlite3", 0, 0) be defined as constants somewhere?

Comment (Karsten1987) [reply rosbag2_cpp/src/rosbag2_cpp/reader.cpp:42]: the last two are directly "hardcoded" in the storage options struct - meaning I could technically leave out the following two lines of code. 
Where would you define the other constant, "sqlite3"?

**Comment by Karsten1987** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11590/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6731/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9469/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11552/)

**Comment by Karsten1987** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11604/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6743/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9487/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11567/)

File: rosbag2_cpp/src/rosbag2_cpp/reader.cpp (hunk 1 above)
Comment (Karsten1987) [reply rosbag2_cpp/src/rosbag2_cpp/reader.cpp:42]: @emersonknapp I've excluded the `sqlite3` into a constant within the same file. Let me know if that works for you.

Comment (emersonknapp) [reply rosbag2_cpp/src/rosbag2_cpp/reader.cpp:42]: LGTM

**Pull request merged**